.buildkite/test-sample-yamls.yml

```diff
@@ -14,11 +14,12 @@
     # Run sample YAML tests
     - echo "--- START:Running Sample YAMLs (nightly operator) tests"
     - set -o pipefail
-    - KUBERAY_TEST_TIMEOUT_SHORT=1m KUBERAY_TEST_TIMEOUT_MEDIUM=5m KUBERAY_TEST_TIMEOUT_LONG=10m go test -timeout 30m -v ./test/sampleyaml 2>&1 | awk -f ../.buildkite/format.awk
+    - KUBERAY_TEST_TIMEOUT_SHORT=1m KUBERAY_TEST_TIMEOUT_MEDIUM=5m KUBERAY_TEST_TIMEOUT_LONG=10m go test -timeout 30m -v ./test/sampleyaml 2>&1 | awk -f ../.buildkite/format.awk || (echo "--- Dumping the Failed test's K8s cluster state" && kubectl get pods,jobs,svc --all-namespaces -o yaml)
```

Comment:
All RayClusters created by tests are ephemeral. Can we capture the failed tests if we add the command here?

Reply:
@kevin85421 Thanks for the review. Good point, it seems make more sense to implement this by a go helper function after test failed.